src/client_side.cc

```diff
@@ -1638,7 +1638,10 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
 #endif
 
     if (internalCheck(request->url.path())) {
-        if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
+        const bool allowLocalhost = (conn->port->s.isLocalhost() || conn->port->s.isAnyAddr());
```

Comment:
Please avoid "allow" or "deny" in variable names that do not contain the final allow/deny decision because such naming makes the code more difficult to understand.

```suggestion
        const auto receivedAtLocalhost = (conn->port->s.isLocalhost() || conn->port->s.isAnyAddr());
```

Reply:
Done in [f1015fa](https://github.com/squid-cache/squid/pull/1406/commits/f1015fa9f7ccf7c9d8122f534ba5bd40b03b5a3f)